test/object-store/sync/app.cpp

```diff
@@ -708,6 +708,83 @@ TEST_CASE("app: link_user integration", "[sync][app]") {
     }
 }
 
+// MARK: - Delete User Tests
+
+TEST_CASE("app: delete anonymous user integration", "[sync][app]") {
+    TestSyncManager tsm(get_integration_config(), {});
+    auto app = tsm.app();
+
+    SECTION("delete user expect success") {
+        CHECK(app->sync_manager()->all_users().size() == 0);
+
+        // Log in user 1
+        auto user_a = log_in(app);
+        CHECK(user_a->state() == SyncUser::State::LoggedIn);
+        app->delete_user(user_a, [&](Optional<app::AppError> error) {
+            REQUIRE_FALSE(error);
+            // a logged out anon user will be marked as Removed, not LoggedOut
+            CHECK(user_a->state() == SyncUser::State::Removed);
+        });
+        CHECK(app->sync_manager()->all_users().empty());
+        CHECK(app->sync_manager()->get_current_user() == nullptr);
+
+        app->delete_user(user_a, [&](Optional<app::AppError> error) {
+            CHECK(error->message == "User must be logged in to be deleted.");
+            CHECK(app->sync_manager()->all_users().size() == 0);
+        });
+
+        // Log in user 2
+        auto user_b = log_in(app);
+        CHECK(app->sync_manager()->get_current_user() == user_b);
+        CHECK(user_b->state() == SyncUser::State::LoggedIn);
+        CHECK(app->sync_manager()->all_users().size() == 1);
+
+        app->delete_user(user_b, [&](Optional<app::AppError> error) {
+            REQUIRE_FALSE(error);
+            CHECK(app->sync_manager()->all_users().size() == 0);
+        });
+
+        CHECK(app->sync_manager()->get_current_user() == nullptr);
+
+        // check both handles are no longer valid
+        CHECK(user_a->state() == SyncUser::State::Removed);
+        CHECK(user_b->state() == SyncUser::State::Removed);
+    }
+}
+
+TEST_CASE("app: delete user with credentials integration", "[sync][app]") {
+    TestSyncManager tsm(get_integration_config(), {});
+    auto app = tsm.app();
+
+    SECTION("log in and delete") {
+        CHECK(app->sync_manager()->all_users().size() == 0);
```

Comment:
could you add some checks for the state of `app->current_user()` around this test?